### PR TITLE
src: pomodoro: Add watch flag to pomodoro

### PR DESCRIPTION
**Pull request by lucasqml**
This pull request addresses the enhancement request detailed in issue [#1115](https://github.com/kworkflow/kworkflow/issues/1115) 

The kw pomodoro functionality has been improved by adding a new --watch option, allowing users to monitor their active pomodoro sessions in real-time.

**Implementation Details**
The kw pomodoro command now accepts a --watch flag.
When --watch is enabled, the command enters a loop, refreshing the display every second to show the current status of the active pomodoro session.
This continuous update mechanism allows users to monitor their pomodoro sessions without manual intervention.

**Benefits**
Improved User Experience: Users can now easily track their pomodoro sessions without re-running commands.
Real-Time Monitoring: Provides a live view of the session progress, enhancing productivity and focus.

**Code Changes**
Updated the kw pomodoro command to include the --watch option.
Implemented a loop that refreshes the session status every second when --watch is enabled.

**Example Usage**
To use the new feature, after setting a timer, run:

`kw pomodoro --watch`

This will display the active pomodoro session and update the status every second in the terminal.

Please review the changes and provide any feedback. Thank you for considering this enhancement to the kw pomodoro functionality.

### CODE REVIEW

File: src/_kw
Comment (rodrigosiqueira) [inline src/_kw:1]: Please expand your commit message to explain why and how you implemented this feature.

https://kworkflow.org/content/howtocontribute.html#commit-messages

File: src/_kw [hunk 1]
```diff
@@ -432,6 +432,7 @@ _kw_pomodoro()
     '(-t --set-timer -c --check-timer -s --show-tags)'{-t,--set-timer}'[set the timer for the Pomodoro timebox]: : ' \
     '(-c --check-timer -t --set-timer -s --show-tags)'{-c,--check-timer}'[shows information of each active Pomodoro timebox]' \
     '(-s --show-tags -c --check-timer -t --set-timer)'{-s,--show-tags}'[shows registered tags]' \
+    '(-w --watch)'{-w,--watch}'[lock the terminal and display the current Pomodoro every second]' \
```
Comment (rodrigosiqueira) [inline src/_kw:435]: you also need to add new arguments to the bash_autocomplete file.

File: src/pomodoro.sh [hunk 2]
```diff
@@ -49,6 +49,16 @@ function pomodoro_main()
   if [[ -n "${options_values['TIMER']}" ]]; then
     timer_thread "$flag" &
   fi
+
+  if [[ -n "${options_values['WATCH']}" ]]; then
+    clear
+    while true; do
+      tput cup 0 0
+      show_active_pomodoro_timebox "$flag"
+      sleep 1
```
Comment (rodrigosiqueira) [inline src/pomodoro.sh:58]: Since you are locking the terminal, it would be nice to handle the Ctrl + C scenario in which the user cancels this operation.

File: src/pomodoro.sh [hunk 3]
```diff
@@ -49,6 +49,16 @@ function pomodoro_main()
   if [[ -n "${options_values['TIMER']}" ]]; then
     timer_thread "$flag" &
   fi
+
+  if [[ -n "${options_values['WATCH']}" ]]; then
+    clear
```
Comment (rodrigosiqueira) [inline src/pomodoro.sh:54]: Move this implementation to its own function.

File: src/pomodoro.sh [hunk 4]
```diff
@@ -305,8 +315,8 @@ function format_text()
 
 function parse_pomodoro()
 {
-  local long_options='set-timer:,check-timer,show-tags,tag:,description:,help,verbose'
-  local short_options='t:,c,s,g:,d:,h'
+  local long_options='set-timer:,check-timer,show-tags,tag:,description:,help,verbose,watch'
+  local short_options='t:,c,s,g:,d:,h,v,w'
```
Comment (rodrigosiqueira) [inline src/pomodoro.sh:319]: Why did you add v here?

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]
Hi,
Thanks for your PR. I added some comments, but for the next version I also want to see:
1. Documentation
2. Unit tests

**Comment by bolgheroni** [comment]
> Hi, Thanks for your PR. I added some comments, but for the next version I also want to see:
> 
>     1. Documentation
> 
>     2. Unit tests

Hi!

Thank you for the review! I'll start working on the requested changes.

File: src/pomodoro.sh (hunk 2 above)
Comment (bolgheroni) [reply src/pomodoro.sh:58]: Thanks for noticing that! I realized that when Ctrl+C is executed, the timer thread is killed too. That is not the desired behavior right?

File: src/pomodoro.sh [hunk 5]
```diff
@@ -49,6 +49,28 @@ function pomodoro_main()
...
     timer_thread "$flag" &
   fi
+
+  if [[ -n "${options_values['WATCH']}" ]]; then
+    watch_loop "$flag"
+    return 0
+  fi
+}
+
+function watch_loop()
```
Comment (rodrigosiqueira) [inline src/pomodoro.sh:59]: Add a function documentation.

File: src/pomodoro.sh [hunk 6]
```diff
@@ -49,6 +49,28 @@ function pomodoro_main()
...
+{
+  local flag="$1"
+
+  clear
+  while true; do
```
Comment (rodrigosiqueira) [inline src/pomodoro.sh:64]: Add a new line between these two lines.

File: src/pomodoro.sh [hunk 7]
```diff
@@ -49,6 +49,28 @@ function pomodoro_main()
...
+    tput cup 0 0
+    show_active_pomodoro_timebox "$flag"
+    echo 'Press Q to quit'
+    read -r -n 1 -t 1 key
+    if [[ "$key" == 'q' || "$key" == 'Q' ]] ; then
+      return 0
+    fi
+  done
+  return 0
```
Comment (rodrigosiqueira) [inline src/pomodoro.sh:73]: Add a new line between these two lines.

File: src/pomodoro.sh [hunk 8]
```diff
@@ -49,6 +49,28 @@ function pomodoro_main()
...
+  fi
+}
+
+function watch_loop()
+{
+  local flag="$1"
+
+  clear
+  while true; do
+    tput cup 0 0
```
Comment (rodrigosiqueira) [inline src/pomodoro.sh:65]: Use cmd_manager; this can help you implement a test.

File: src/pomodoro.sh [hunk 9]
```diff
@@ -49,6 +49,28 @@ function pomodoro_main()
...
+    watch_loop "$flag"
+    return 0
+  fi
+}
+
+function watch_loop()
+{
+  local flag="$1"
+
+  clear
```
Comment (rodrigosiqueira) [inline src/pomodoro.sh:63]: Use cmd_manager; this can help you implement a test.

File: src/pomodoro.sh [hunk 10]
```diff
@@ -49,6 +49,28 @@ function pomodoro_main()
...
+  while true; do
+    tput cup 0 0
+    show_active_pomodoro_timebox "$flag"
+    echo 'Press Q to quit'
```
Comment (rodrigosiqueira) [inline src/pomodoro.sh:67]: https://kworkflow.org/content/codingstyle.html#use-printf-instead-of-echo

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]
Hi,

I added some comments. Additionally, you should add tests, code, and user docs for your change.

Also, I don't see any real-time update of the timebox going down if I just use `kw p -t 1m` followed by `kw p -w` or even with `kw p -t 1m -w`. Is this what you want? It just print the message 'Press Q to quit'.

**Comment by bolgheroni** [comment]
Hi!

Thanks for the comments. I'm still working on some of the changes. 

About the following issue:

> Also, I don't see any real-time update of the timebox going down if I just use `kw p -t 1m` followed by `kw p -w` or even with `kw p -t 1m -w`. Is this what you want? It just print the message 'Press Q to quit'.

Thanks for noticing it, I believe this was due the session not having a tag set. Running `kw p --check-timer` apparently also prints nothing.

**Comment by bolgheroni** [comment]
I was actually struggling with properly handling the `Ctrl+C` command.  I tried using `signal_manager`, like in the following example, but the "default" handler kept executing, and the timer thread was killed. 

```shell
function exit_gracefully()
{
  exit 0
}

function watch_loop()
{
  local flag="$1"
  clear
  while true; do
    signal_manager 'exit_gracefully' SIGINT SIGTERM
    tput cup 0 0
    show_active_pomodoro_timebox "$flag"
    echo 'Press Q to quit'
    read -r -n 1 -t 1 key
    if [[ "$key" == 'q' || "$key" == 'Q' ]] ; then
      return 0
    fi
  done
  return 0
}
```

**Comment by bolgheroni** [comment]
Hi!

This last change I've pushed contains some corner cases handling, such as starting the watch mode without tagged timers and handling the timer finishing.
I'm still working on adding unit tests and documentation.

**Comment by rodrigosiqueira** [comment]
@bolgheroni could you rebase this PR and push it with documentation and test?

**Comment by bolgheroni** [comment]
@rodrigosiqueira yeah, absolutely! I was wondering where should we add documentation. 
I know there's the [man page](https://kworkflow.org/man/features/kw-pomodoro.html), and I've also seen that `pomodoro` has a [tutorial](https://kworkflow.org/tutorials/pomodoro-report.html).

**Comment by rodrigosiqueira** [comment]
Hi @bolgheroni , I closed this PR due to the lack of interactions. If you have a new version that addresses the comments of the previous reviews, feel free to open a new PR.
